MGMT-10281: Add documentation for the converged flow (#3873)
openshift/assisted-service: docs/hive-integration/README.md

    - Hostname (optional for user to set)
    - MachineConfigPool (optional for user to set)
- Reconcile the BareMetalHost hardware details by copying the Agent's inventory data to the BMH's `hardwaredetails` annotation.

> - Disable ironic inspection disabled

Reviewer: Failed to parse - did you mean:
```suggestion
- Disable ironic inspection
```

See BMAC documentation [here](./baremetal-agent-controller.md).

## Bare Metal Operator Integration using the converged flow
See reference [here](https://github.com/openshift/enhancements/blob/master/enhancements/baremetal/ztp-metal3.md)

> In case the Bare Metal Operator on the HUB cluster version is 4.11 or higher, the integration between the MBO and the assisted-service is slightly different.

Reviewer 2: bmo?

Author: Bare Metal Operator AKA: BMO

Reviewer 2: Metal Bare Operator AKA: MBO

> The assisted-service will run a Preprovisioning Image Controller that reconcile the PreprovisioningImage for the  BareMetalHost.

Reviewer: ```suggestion
The assisted-service will run a Preprovisioning Image Controller that reconciles PreprovisioningImages and:
```

> - Annotate the InfraEnv for the BareMetalHost with `infraenv.agent-install.openshift.io/enable-ironic-agent="true"`
> - Add the ironic agent config to the discovery ignition embeded to the InfraEnv ISO
> - Copy the InfraEnv ISODownloadURL to the PreprovisioningImage ImageUrl
>   This allows the host to register with metal3 and let metal3 reconcile the BareMetalHost hardware details.

Reviewer: ```suggestion
- Annotates the InfraEnv for the BareMetalHost with `infraenv.agent-install.openshift.io/enable-ironic-agent="true"`
- Adds the ironic agent config to the discovery ignition embedded in the InfraEnv ISO
- Copies the InfraEnv ISODownloadURL to the PreprovisioningImage ImageUrl, which allows the host to register with metal3 and let metal3 reconcile the BareMetalHost hardware details.
```

> The Baremetal Agent Controller will will stop explicitly disabling *inspection* on `BareMetalHost`s it manages

Reviewer 2: will will will

and will no longer update the BareMetalHost with detached annotation.
It will continue to:
- Find the right pairs of BMH/Agent using their MAC addresses
- Reconcile the Agent's spec by copying the following attributes from the BMH's annotations:
  - Role: master/worker
  - Hostname (optional for user to set)
  - MachineConfigPool (optional for user to set)
- Set the  BareMetalHost CustomDeploy Method to `start_assisted_install` - this custom deploy method will be added to start
  the assisted agent when BMO requests deployment.

The assisted agent will not reboot the machine at the end of the installation, instead it will stop

> the assisted agent service and let the ironic agent to manage the machine power state

Reviewer 2: just to verify - ai will write all the coreos-install flags before the reboot, right? It's only the reboot that is missing

Reviewer: Yes

Author: Right.
Also, note that the converged flow preserves the host installer args overrides and ignition config overrides functionality (annotations on the BMH are copied to the agent)

    - Hostname (optional for user to set)
    - MachineConfigPool (optional for user to set)
- Reconcile the BareMetalHost hardware details by copying the Agent's inventory data to the BMH's `hardwaredetails` annotation.
- Disable ironic inspection

See BMAC documentation [here](./baremetal-agent-controller.md).

> ## Bare Metal Operator Integration using the converged flow

Reviewer 2: Lets document disks cleanup flow?
